### PR TITLE
fix(l1): add temporary fix to transaction filtering & handle tx filtering unwrap edge case

### PR DESCRIPTION
**Pull request by ilitteri**
**Motivation**

To avoid panicking when the transaction filter fails to filter a transaction because we do not yet filter transactions with invalid tip.

**Description**

- Filter transactions whose effective gas tip calculation underflowed.
- Handle the unwrap of the `effective_gas_tip` calculation. Even though this is an edge case that shouldn't happen, it is better to have a descriptive error if it happens someday than to panic.

### CODE REVIEW

**Review by fkrause98 — APPROVED** [review]

**Review by mpaulucci — APPROVED** [review]

File: crates/blockchain/mempool.rs [hunk 1]
```diff
@@ -81,10 +81,7 @@ pub fn filter_transactions(
         }
         // Filter by tip & base_fee
         if let Some(min_tip) = filter.min_tip {
-            if !tx
-                .effective_gas_tip(filter.base_fee)
-                .is_some_and(|tip| tip >= min_tip)
-            {
+            if !tx.effective_gas_tip(filter.base_fee) >= min_tip {
```
Comment (rodrigo-o) [inline crates/blockchain/mempool.rs:84]: np: maybe `if tx.effective_gas_tip(filter.base_fee) < min_tip {`

**Review by rodrigo-o — COMMENTED** [review]
LGTM (small np comment that could be ignored)

**Review by rodrigo-o — APPROVED** [review]

File: crates/blockchain/mempool.rs (hunk 1 above)
Comment (ilitteri) [reply crates/blockchain/mempool.rs]: Thanks! Done [49eac70](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1018/commits/49eac7011cb90d8c55f9d52bb7edc35c4d4d5489)

File: crates/blockchain/error.rs [hunk 2]
```diff
@@ -32,6 +32,10 @@ pub enum InvalidBlockError {
     GasUsedMismatch,
     #[error("Blob gas used doesn't match value in header")]
     BlobGasUsedMismatch,
+    #[error(
+        "Attempted to add and invalid transaction to the block. The transaction filter must have been failed."
```
Comment (fmoletta) [inline crates/blockchain/error.rs:36]:
```suggestion
        "Attempted to add an invalid transaction to the block. The transaction filter must have failed."
```

Comment (ilitteri) [reply crates/blockchain/error.rs:36]: Thanks!

File: crates/blockchain/mempool.rs [hunk 3]
```diff
@@ -79,6 +79,14 @@ pub fn filter_transactions(
         if filter.only_plain_txs && is_blob_tx || filter.only_blob_txs && !is_blob_tx {
             return false;
         }
+
+        // This is a temporary fix to avoid invalid transactions to be included.
+        // This should be removed once https://github.com/lambdaclass/ethereum_rust/issues/680
+        // is addressed.
+        if tx.effective_gas_tip(filter.base_fee).is_none() {
+            return false;
+        }
```
Comment (fmoletta) [inline crates/blockchain/mempool.rs:88]: We could add this check on the `else` block of the min_tip check, that way we can leave the check even when the issue is fixed

Comment (ilitteri) [reply crates/blockchain/mempool.rs:88]: But, this should be an `else if` right? We still need to know if `effective_gas_tip` returned `None`, or am I missing something?

Comment (rodrigo-o) [reply crates/blockchain/mempool.rs:88]: As I understood it, yes, we still need to check for the `None` of `effective_gas_tip` even in the else of `let Some(min_tip) = filter.min_tip`

File: crates/blockchain/error.rs [hunk 4]
```diff
@@ -32,6 +32,10 @@ pub enum InvalidBlockError {
     GasUsedMismatch,
     #[error("Blob gas used doesn't match value in header")]
     BlobGasUsedMismatch,
+    #[error(
+        "Attempted to add an invalid transaction to the block. The transaction filter must have failed."
+    )]
+    AttemptedToAddInvalidTransaction,
```
Comment (mpaulucci) [inline crates/blockchain/error.rs:38]: the error is a bit long. Why not just `InvalidTransaction`?

File: crates/blockchain/mempool.rs (hunk 3 above)
Comment (ilitteri) [reply crates/blockchain/mempool.rs:88]: Gotcha

File: crates/blockchain/error.rs (hunk 4 above)
Comment (ilitteri) [reply crates/blockchain/error.rs:38]: Not a problem, renaming it asap.

File: crates/blockchain/mempool.rs (hunk 3 above)
Comment (ilitteri) [reply crates/blockchain/mempool.rs]: Done [62e4d9c](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1018/commits/62e4d9c11fe53c6cb79bc58192b818c304cd6f01)

File: crates/blockchain/error.rs (hunk 4 above)
Comment (ilitteri) [reply crates/blockchain/error.rs]: Done [6f3a2fb](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1018/commits/6f3a2fbdf00a4ad6a87cfac57271f09a27d7d2ef)

**Review by fmoletta — APPROVED** [review]

**Pull request merged**